src/main/java/nl/altindag/log/LogCaptor.java

```diff
@@ -124,20 +135,24 @@ public static LogCaptor forName(String name) {
     }
 
     private List<String> getLogs(Level level) {
-        return listAppender.list.stream()
-                .filter(logEvent -> logEvent.getLevel() == level)
-                .map(ILoggingEvent::getFormattedMessage)
-                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
+        synchronized (eventList) {
+            return eventList.stream()
+                    .filter(logEvent -> logEvent.getLevel() == level)
+                    .map(ILoggingEvent::getFormattedMessage)
+                    .collect(collectingAndThen(toList(), Collections::unmodifiableList));
+        }
     }
 
     public List<LogEvent> getLogEvents() {
-        return listAppender.list.stream()
-                .map(toLogEvent())
-                .collect(toList());
+        synchronized (eventList) {
+            return eventList.stream()
+                    .map(toLogEvent())
+                    .collect(toList());
```

Comment:
Would it be possible for you to refactor this line to `.collect(collectingAndThen(toList(), Collections::unmodifiableList));` It looks like I have overseen this part when I worked on it